File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/repositories/AdjustmentsRepository.java
package me.mxcsyounes.archernotebook.repositories;

import android.app.Application;
import android.arch.lifecycle.LiveData;

import java.util.List;

import me.mxcsyounes.archernotebook.asynck.AdjDatabaseTask;
import me.mxcsyounes.archernotebook.database.ArcherDatabase;
import me.mxcsyounes.archernotebook.database.dao.AdjustmentDao;
import me.mxcsyounes.archernotebook.database.entities.Adjustment;

public class AdjustmentsRepository {

    private AdjustmentDao mAdjDao;
    private LiveData<List<Adjustment>> mAllAdj;

    public AdjustmentsRepository(Application application) {
        mAdjDao = ArcherDatabase.getInstance(application, false).mAdjDao();
        mAllAdj = mAdjDao.getAllDateDESC();
    }

    public LiveData<List<Adjustment>> getAllAdj() {
        return mAllAdj;
    }

    public LiveData<Adjustment> getAdj(int id) {
        return mAdjDao.getAdjustmentById(id);
    }

    public void insertAdjustment(Adjustment adjustment) {
        new AdjDatabaseTask(AdjDatabaseTask.INSERT, mAdjDao).execute(adjustment);
    }

    public void updateNote(Adjustment adjustment) {
        new AdjDatabaseTask(AdjDatabaseTask.UPDATE, mAdjDao).execute(adjustment);
    }

    public void deleteAdjustment(Adjustment adjustment) {
        new AdjDatabaseTask(AdjDatabaseTask.DELETE, mAdjDao).execute(adjustment);
    }

    public void deleteAllAdjustments() {
        new AdjDatabaseTask(AdjDatabaseTask.DELETE_ALL, mAdjDao).execute();
    }
}

File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/repositories/AdjustmentsRepository.kt
package me.mxcsyounes.archernotebook.repositories

import android.app.Application
import android.arch.lifecycle.LiveData
import me.mxcsyounes.archernotebook.database.ArcherDatabase
import me.mxcsyounes.archernotebook.database.dao.AdjustmentDao
import me.mxcsyounes.archernotebook.database.entities.Adjustment
import java.util.concurrent.Executors

class AdjustmentsRepository(application: Application) {

    private val mAdjDao: AdjustmentDao = ArcherDatabase.getInstance(application, false).mAdjDao()
    val allAdj: LiveData<MutableList<Adjustment>>
    private val executor = Executors.newSingleThreadExecutor()

    init {
        allAdj = mAdjDao.getAllDateDESC()
    }

    fun getAdjustment(id: Int): LiveData<Adjustment> {
        return mAdjDao.getAdjustmentById(id)
    }

    fun insertAdjustment(adjustment: Adjustment) {
        executor.execute {
            mAdjDao.insertAdjustment(adjustment)
        }
    }

    fun updateAdjustment(adjustment: Adjustment) {
        executor.execute {
            mAdjDao.updateAdjustment(adjustment)
        }
    }

    fun deleteAdjustment(adjustment: Adjustment) {
        executor.execute {
            mAdjDao.deleteAdjustment(adjustment)
        }
    }

    fun deleteAllAdjustments() {
        executor.execute {
            mAdjDao.deleteAll()
        }
    }
}
